README.md

```diff
@@ -22,25 +22,53 @@ After installing, you need to require Composer's autoloader:
 require 'vendor/autoload.php';
 ```
 
+## Setting up a Request Adapter
+Because of dependency collision, we have opted to use a request adapter rather than
+requiring a request library.  This means that your application will need to pass in
+a request adapter to the constructor of the SparkPost Library.  We use the [Ivory HTTP Adapter] (https://github.com/egeloen/ivory-http-adapter) in SparkPost. Please visit their repo
+for a list of supported adapters.  If you don't currently use a request library, you will
+need to require one and create an adapter from it and pass it along.  The example below uses the
+GuzzleHttp Client Library.
+
+An Adapter can be setup like so:
+```
+use SparkPost\SparkPost;
+use GuzzleHttp\Client;
+use Ivory\HttpAdapter\Guzzle6HttpAdapter;
+
+$httpAdapter = new Guzzle6HttpAdapter(new Client());
+$sparky = new SparkPost($httpAdapter, ['key'=>'YOUR API KEY']);
+```
+
 ## Getting Started:  Your First Mailing
 ```
-SparkPost::setConfig(["key"=>"YOUR API KEY"]);
+use SparkPost\SparkPost;
+use GuzzleHttp\Client;
+use Ivory\HttpAdapter\Guzzle6HttpAdapter;
+
+$httpAdapter = new Guzzle6HttpAdapter(new Client());
+$sparky = new SparkPost($httpAdapter, ['key'=>'YOUR API KEY']);
 
 try {
-	// Build your email and send it!
-	Transmission::send(array('campaign'=>'first-mailing', 
-		'from'=>'you@your-company.com',
-	    'subject'=>'First SDK Mailing',
-	    'html'=>'<html><body><h1>Congratulations, {{name}}!</h1><p>You just sent your very first mailing!</p></body></html>',
-	    'text'=>'Congratulations, {{name}}!! You just sent your very first mailing!',
-	    'substitutionData'=>array('name'=>'YOUR FIRST NAME'),
-	    'recipients'=>array(array('address'=>array('name'=>'YOUR FULL NAME', 'email'=>'YOUR EMAIL ADDRESS' )))
-    ));
-
-    echo 'Woohoo! You just sent your first mailing!';
-} catch (Exception $err) {
-    echo 'Whoops! Something went wrong';
-    var_dump($err);
+	$results = $sparky->transmission->send([
+		'from'=>'From Envelope <from@sparkpostbox.com>',
+		'html'=>'<html><body><h1>Congratulations, {{name}}!</h1><p>You just sent your very first mailing!</p></body></html>',
+		'text'=>'Congratulations, {{name}}!! You just sent your very first mailing!',
+    'substitutionData'=>array('name'=>'YOUR FIRST NAME')
+		'subject'=>'First Mailing From PHP',
+		'recipients'=>[
+  		[
+  			"address"=>[
```

Comment:
I should see if I can't find a php-linter just to enforce some consistency.  If I can, I'll make it another issue.